src/darwin/Framework/CHIP/MTRCommissioningDelegate.h

```diff
@@ -0,0 +1,112 @@
+@optional
+/**
+ * Callback that gets called after various information (product identity,
+ * optionally endpoint structure information, optionally other attributes that
+ * were requested) has been read from the commissionee.
+ */
+- (void)commissioning:(MTRCommissioningOperation *)commissioning
+    readCommissioneeInfo:(MTRCommissioneeInfo *)info;
+
+@required
+/**
+ * Notification that device attestation has completed.
+ *
+ * Commissioning will pause, regardless of whether attestation succeeded or
+ * failed, until the completion is invoked (indicating that commissioning should
+ * proceed, even if attestation failed), or commissioning is stopped.
+ */
+- (void)commissioning:(MTRCommissioningOperation *)commissioning
+    completedDeviceAttestation:(MTRDeviceAttestationDeviceInfo *)attestationDeviceInfo
+                         error:(nullable NSError *)error
+                    completion:(dispatch_block_t)completion;
+
+@optional
+/**
+ * Callback that gets called for a commissionee that supports Wi-Fi if
+ * Wi-Fi network commissioning is required and Wi-Fi credentials were not
+ * provided in MTRCommissioningParameters.
+ *
+ * Commissioning will pause, including in cases when the network scan failed,
+ * until the completion is invoked with the network and credentials the
+ * commissionee should use, or until commissioning is stopped.  The provided
+ * credentials are allowed to be nil for an open Wi-Fi network.
+ *
+ * Exactly one of networks and error will be non-nil.
+ */
+- (void)commissioning:(MTRCommissioningOperation *)commissioning
+    scannedWiFiNetworks:(nullable NSArray<MTRNetworkCommissioningClusterWiFiInterfaceScanResultStruct *> *)networks
+                  error:(nullable NSError *)error
+             completion:(void (^)(NSData * ssid, NSData * _Nullable credentials))completion;
+
+/**
+ * Callback that gets called for a commissionee that supports Thread if Thread
+ * network commissioning is required and an operational dataset was not provided
+ * in MTRCommissioningParameters.
+ *
+ * Commissioning will pause, including in cases when the network scan failed,
+ * until the completion is invoked with the operational dataset the commissionee
+ * should use, or until commissioning is stopped.
+ *
+ * Exactly one of networks and error will be non-nil.
+ */
+- (void)commissioning:(MTRCommissioningOperation *)commissioning
+    scannedThreadNetworks:(nullable NSArray<MTRNetworkCommissioningClusterThreadInterfaceScanResultStruct *> *)networks
+                    error:(nullable NSError *)error
+               completion:(void (^)(NSData * operationalDataset))completion;
+
+/**
+ * Notification that network credentials have been successfully communicated
+ * to the commissionee and it's going to try to join that network.  Note that
+ * for commissionees that are already on-network this notification will not
+ * happen.
+ */
+- (void)commissioningProvisionedNetworkCredentials:(MTRCommissioningOperation *)commissioning;
```

Comment:
We could have some generic "progress" callback that just provides the MTRCommissioningOperation and a step enum?

Reply:
Yeah.... I need to think about what should go in that enum.  But it would probably be a good idea given some other use cases Home has for knowing progress here.

Reply:
Done.